sktime/pipeline/pipeline.py

```diff
@@ -0,0 +1,164 @@
+from copy import deepcopy
+
+from sktime.base import BaseEstimator
+from sktime.pipeline.computation_setting import ComputationSetting
+from sktime.pipeline.step import Step
+from sktime.transformations.series.subset import ColumnSelect
+
+
+class Pipeline(BaseEstimator):
+
+    def __init__(self, step_informations=None):
+        super().__init__()
+        self.computation_setting = ComputationSetting()
+
+        # Initialise the method
+        self.steps = {"X": Step(None, "X", None, {}, compuatation_setting=self.computation_setting),
+                      "y" : Step(None, "y", None, {}, compuatation_setting=self.computation_setting)}
+        self.model_dict = {}
+        if step_informations is not None:
+            for step_info in step_informations:
+                self.add_step(**step_info)
+
+    def _get_unique_id(self, skobject):
+        return -1
+
+    def _get_step(self, name):
+        # TODO do here subsetting using Subsetter
+        #   ColumnSelect Transformer
+        if name in self.steps:
+            return self.steps[name]
+
+        raise Exception("Required Input does not exist")
+
+    def add_step(self, skobject, name, edges, **kwargs):
+        """
+        TODO
+        """
+        unique_id = self._get_unique_id(skobject)
+        if not unique_id in self.model_dict:
+            self.model_dict[unique_id] = skobject.clone()
+
+        input_steps = {}
+        for key, edge in edges.items():
+            edge = edge if isinstance(edge, list) else [edge]
+            for edg in edge:
+                if "__" in edg and not edg in self.steps:
+                    # Just semantic sugar..
+                    self._create_subsetter(edg)
+                input_steps[key] = [self._get_step(edg) for edg in edge]
+
+        step = Step(skobject, name, input_steps, kwargs, compuatation_setting=self.computation_setting)
+        if name in self.steps:
+            raise Exception("Name Conflict")
+
+        self.steps[name] = step
+        self._last_step_name = name
+        return step
+
+
+    def fit(self, X, y=None, **kwargs):
+        # Fits the pipeline
+        self.steps["X"].buffer = X
+        self.steps["y"].buffer = y
+
+        self.computation_setting.method_resolution_order = ["transform", "predict"]
```

Comment:
is this sth we need to hardcode in the `computation_setting`? I'd rather do dispatch on type or similar, but not sure if that has domino effects

Reply:
I have introduced the method_resolution_order due to the `predict_interval` method and the assumption that there might be a forecaster that is only deterministic, and by specific postprocessing steps, the forecast become probabilistic.
But perhaps using dispatching is better and more general. I need to think about it